contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8043,48 +8082,98 @@ rewrite_information_schema_to_information_schema_tsql(T ctx, GetCtxFunc<T> getSc
 /* if no rewriting necessary, return empty string */
 template<class T>
 std::string
-rewrite_object_name_with_omitted_db_and_schema_name(T ctx, GetCtxFunc<T> getDatabase, GetCtxFunc<T> getSchema)
+rewrite_object_name_with_omitted_db_and_schema_name(T ctx, GetCtxFunc<T> getDatabase, GetCtxFunc<T> getSchema, GetCtxFunc<T> getObject)
 {
+	auto schema = getSchema(ctx);
+	std::string objName = "";
+	bool schema_is_dbo_or_sys = false;
+	bool catalog_need_sys_schema = false;
+	bool must_rewrite = false; // Handles case of .dbname..sometable for a non-catalog table
+
+	if (getObject != nullptr)
+	{
+		std::string objNameStripped = "";
+		objName = ::getFullText(getObject(ctx));
+		objNameStripped = stripQuoteFromId(objName);
```

Comment:
The `GetCtxFunc<T> getObject` argument (which was added in this fix) resolves to different object types depending on which parser rule this is invoked from. That does not match with the original declaration of stripQuoteFromId(), so something had to be changed. Adding another copy of this proc with a string argument was the least amount of change since stripQuoteFromId() is already invoked from 30+ locations.